Connection.js

```diff
@@ -413,15 +413,48 @@ Connection.prototype.answerHandshake = function (lines) {
 
 	this.key = this.headers['sec-websocket-key']
 
+	// Read protocols
+	var protocols = this.headers['sec-websocket-protocol']
+
+	if (typeof protocols !== "undefined") {
+		this.requestedProtocols = protocols
+			.split(",")
+			.map(function(value) {
+				return value.trim()
+			})
+	}
+
 	// Build and send the response
 	sha1 = crypto.createHash('sha1')
 	sha1.end(this.key + '258EAFA5-E914-47DA-95CA-C5AB0DC85B11')
 	key = sha1.read().toString('base64')
-	this.socket.write(this.buildRequest('HTTP/1.1 101 Switching Protocols', {
+
+	response = {
 		Upgrade: 'websocket',
 		Connection: 'Upgrade',
 		'Sec-WebSocket-Accept': key
-	}))
+	}
+
+	// Choose a protocol
+	if (this.server && this.server.acceptedSubprotocols) {
+		var acceptedProtocols = this.server.acceptedSubprotocols
+		var requestedProtocols = this.requestedProtocols
+
+		// The earlier a protocol appears in our accepted list, the higher its priority
+		var firstProtocol = acceptedProtocols.length - 1
+		requestedProtocols.forEach(function (protocol) {
```

Comment:
The spec does not demand any specific algorithm for selecting the protocol to use.

It says the client list if ordered by preference and the server should pick *at most* one of them.
But it does not hint which preference should be honored when there are more than one common protocol.

For example, if the client sends `Sec-WebSocket-Protocol: a, b, c` and the server has `acceptedProtocols = ['c', 'a', 'd']`. Should we honor client's preference for `a` or server's for `c`?

I'm more inclined to follow the client's preference, because the spec explicitly says the client has a preference.